cpp/src/Slice/Grammar.y

```diff
@@ -1765,112 +1782,81 @@ enumerator_initializer
 ;
 
 // ----------------------------------------------------------------------
-out_qualifier
+parameter
 // ----------------------------------------------------------------------
-: ICE_OUT
-{
-    auto out = make_shared<BoolTok>();
-    out->v = true;
-    $$ = out;
-}
-| %empty
+: optional_type_id
 {
-    auto out = make_shared<BoolTok>();
-    out->v = false;
-    $$ = out;
-}
-;
+    auto tsp = dynamic_pointer_cast<OptionalDefTok>($1);
```

Comment:
I notice you didn't change this name, but`tsp` is quite cryptic, what does it stand for?

Reply:
My guess has always been "type string pointer".
But no, you're right. We really love using cryptic names in the grammar.
Maybe I'll do a quick cleanup-pass next week to make these all readable.

Reply:
> My guess has always been "type string pointer".

Nice one!